model/operation.cc

```diff
@@ -250,9 +250,9 @@ namespace minsky
         float l=OperationBase::l*z, r=OperationBase::r*z, 
           h=OperationBase::h*z;
           
-        if (fabs(l)<0.5*iWidth()*z) l=-0.5*iWidth()*z;        
-        if (r<0.5*iWidth()*z) r=0.5*iWidth()*z;    
-        if (h<0.5*iHeight()*z) h=0.5*iHeight()*z;    
+		if (fabs(l)!=iWidth()*z && iWidth()>0) l=-iWidth()*z;        
+		if (r!=iWidth()*z && iWidth()>0) r=iWidth()*z;    
+		if (h!=iHeight()*z && iHeight()>0) h=iHeight()*z;   
```

Comment:
First clause of condition not needed (r!=iWidth()*z) Assuming l is always -ve, then it is not needed on line 253 too.